### ISSUE TITLE
23.12 nightly wheels missing Python minor version

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/cucim#641]
Currently the cuCIM 23.12 nightly wheels are showing up as if they are Python 3 universal wheels. Please see the screenshots below

<details>
<summary><code>cucim-cu12</code>:</summary>

<img width="1840" alt="Screenshot 2023-11-28 at 5 50 54 PM" src="https://github.com/rapidsai/cucim/assets/3019665/c20b2edf-e797-4f9a-9e7e-03979706108a">

</details>

<details>
<summary><code>cucim-cu11</code>:</summary>

<img width="1840" alt="Screenshot 2023-11-28 at 5 50 40 PM" src="https://github.com/rapidsai/cucim/assets/3019665/72b23731-a72a-430e-b653-bef5738d4c9f">

</details>

However they have strict Python version dependence, which is not being reflected

Basically this is a consequence of the wheels being treated as pure when they are not

Would be good to correct this somehow

**Comment by jakirkham** [comment]
This [SO answer]( https://stackoverflow.com/a/75240449 ) gave a nice summary of potential options
